script.js

```diff
@@ -1,5 +1,202 @@
+    return myOutputValue
+  }
+  if (currentGameMode == "Enter Player 2 name"){
+    player2 = input
+    currentGameMode = "Player 1 turn"
+    console.log(`Player 2: ${player1}`);
+    myOutputValue = `Welcome ${player2}. <br> Player 1, press enter to start your roll `;
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  // 
+  var choiceErrorMsg = function(playerNum){
+    if(playerNum == 1){
+      return `Please enter your choice. <br><br> 
+      dice 1 : ${player1diceResults[0]} <br><br>
+      dice 2 : ${player1diceResults[1]} <br><br>
+      Please choose the order of desired dice result. <br>
+      Type 1 and click enter if you wish for dice 1 to be the first number <br>
+      Type 2 and click enter if you wish for dice 2 to be the first number`}
+    else if (playerNum == 2){
+      return `Please enter your choice. <br><br> 
+      dice 1 : ${player2diceResults[0]} <br><br>
+      dice 2 : ${player2diceResults[1]} <br><br>
+      Please choose the order of desired dice result. <br>
+      Type 1 and click enter if you wish for dice 1 to be the first number <br>
+      Type 2 and click enter if you wish for dice 2 to be the first number`
+    }
+  }
+
+  // Start player's turn
+  // Player will output dice result
+  // Player will choose which dice to put first
+  // ^Create shared function to do 2 things to be reused for player 2
+  if(currentGameMode == "Player 1 turn"){
+    myOutputValue = playerTurn(1)
+    currentGameMode = "Player 1 choice"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  if(currentGameMode == "Player 1 choice" && input != ""){
+    player1DiceOrder = input
+    myOutputValue = playerTurn(1, player1DiceOrder,player1diceResults)
+    currentGameMode = "Player 2 turn"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  else if(currentGameMode == "Player 1 choice" && input == ""){
+    currentGameMode = "Player 1 choice"
+    return choiceErrorMsg(1)
+  }
+  if(currentGameMode == "Player 2 turn"){
+    myOutputValue = playerTurn(2)
+    currentGameMode = "Player 2 choice"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  if(currentGameMode == "Player 2 choice" && input != ""){
+    player2DiceOrder = input
+    myOutputValue = playerTurn(2, player2DiceOrder,player2diceResults)
+    currentGameMode = "Game Outcome"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  else if(currentGameMode == "Player 2 choice" && input == ""){
+    currentGameMode = "Player 2 choice"
+    return choiceErrorMsg(2)
+  }
+  if (currentGameMode == "Game Outcome"){
+    myOutputValue = gameOutcome()
+    console.log(`currentGameMode: ${currentGameMode}`);
+    currentGameMode = "Player 1 turn"
+    // reinitialise array data
+    player1diceResults = []
+    player2diceResults = []
+    return myOutputValue
+  }
```

Comment:
again nice use of a helper function! your main function currently is kinda huge, but my suggestion above on refactoring the code should alleviate it somewhat. for separate if statements such as this one, consider putting line breaks also to make it more readable and clear that its a separate section of the function altogether

Reply:
also your resetting of the game state is fine here since you are only resetting the two dice results array. but consider having a function to reset the game state if you have more state variables to reset, which might be the case for blackjack